src/librustdoc/doctest.rs

```diff
@@ -310,7 +310,10 @@ fn run_test(
     let (test, line_offset, supports_color) =
         make_test(test, Some(crate_name), lang_string.test_harness, opts, edition, Some(test_id));
 
-    let output_file = outdir.path().join("rust_out");
+    // Make sure we emit well-formed executable names for our platform.
+    let mut output_file = outdir.path().join("rust_out");
+    output_file.set_extension(env::consts::EXE_EXTENSION);
```

Comment:
This should read from the target spec instead I think. Rustdoc allows running tests for a different target than the host using `--runtool`. This wouldn't fix running windows tests using wine or wasm tests using a wasm runtime I think.

Reply:
Ah, good catch.

Reply:
Do you have an example of such a `--runtool` based workflow I can run on an x86_64-unknown-linux-gnu host? I have drafted the patch to handle the cross compilation case but I want to make sure it works.